### PR TITLE
Add: Create User Model Class (Fixes #7)

### LINKED ISSUE #7: Client: Model - User
Create a frontend User model. 

I wonder if it's possible to leverage the Mongoose model that we already created - maybe there's a piece of that (the configuration object containing the schema) that can be extracted out of the Mongoose schema definition and used in both the backend and frontend. A quick investigation should tell us if that's possible.

### PR DESCRIPTION
**Pull request by mjayjones13**

### CODE REVIEW

File: lib/server/db/model/user.js [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+userSchema.path('lastName').validate(validator.notEmpty(), 'Please provide a last name.');
+userSchema.path('username').validate(validator.notEmpty(), 'Please provide a username.');
+userSchema.path('password').validate(validator.notEmpty(), 'Please provide a password.');
+
+// compile our model
+var User = mongoose.model("User", userSchema);
+
+// Virtual property getter & setter for user.fullName
+userSchema.virtual('fullName').
+    get(function () { return this.firstName + ' ' + this.lastName; }).
```
Comment (joshcox) [inline lib/server/db/model/user.js]: When chaining methods together, JavaScript standard is to place the dot on the same line as the chained method:

```
userSchema.virtual("fullName")
     .get()
     .set();
```

**Review by joshcox — COMMENTED** [review]
Just added one comment.

File: lib/server/db/model/user.js (hunk 1 above)
Comment (mjayjones13) [reply lib/server/db/model/user.js]: Fixed

File: lib/server/db/model/user.js [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+var userSchema = new mongoose.Schema({
+    firstName: { type: String, required: true },
+    lastName:  { type: String, required: true },
+    username:  { type: String, required: true },
+    password:  { type: String, required: true }
+
+});
+
+// Validations
+userSchema.path('firstName').validate(validator.notEmpty(), 'Please provide a first name.');
```
Comment (joshcox) [inline lib/server/db/model/user.js]: [Document#validate](http://mongoosejs.com/docs/api.html#document_Document-validate) takes a function as it's second argument.

Our function should take in an `err` parameter. If the `err` value is truthy, we should log out the error to the console.

**Review by joshcox — APPROVED** [review]
+1 LGTM

**Pull request merged**